docs/proposals/dedicated-nodes.md

```diff
@@ -0,0 +1,294 @@
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+
+<h2>PLEASE NOTE: This document applies to the HEAD of the source tree</h2>
+
+If you are using a released version of Kubernetes, you should
+refer to the docs that go with that version.
+
+<strong>
+The latest release of this document can be found
+[here](http://releases.k8s.io/release-1.1/docs/proposals/dedicated-nodes.md).
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+# Taints, Tolerations, and Dedicated Nodes
+
+## Introduction
+
+This document describes *taints* and *tolerations*, which constitute a generic mechanism for restricting
+the set of pods that can use a node. We also describe one concrete use case for the mechanism,
+namely to limit the set of users (or more generally, authorization domains)
+who can access a set of nodes (a feature we call
+*dedicated nodes*). There are many other uses--for example, a set of nodes with a particular
+piece of hardware could
+be reserved for pods that require that hardware, or a node could be marked as inaccessible
+to all pods when it is being drained before shutdown; see #17190 for more discussion.
+
+## Taints, tolerations, and dedicated nodes
+
+A *taint* is a new type that is part of the `NodeSpec`; when present, it prevents pods
+from scheduling onto the node unless the pod *tolerates* the taint (tolerations are listed
+in the `PodSpec`). Note that there are actually multiple flavors of taints: taints that
+prevent scheduling on a node, taints that cause the scheduler to try to avoid scheduling
+on a node but do not prevent it, taints that prevent a pod from starting on Kubelet even
+if the pod's `NodeName` was written directly (i.e. pod did not go through the scheduler),
+and taints that evict already-running pods.
+[This comment](https://github.com/kubernetes/kubernetes/issues/3885#issuecomment-146002375)
+has more background on these diffrent scenarios. We will focus on the first
+kind of taint in this doc, since it is the kind required for the "dedicated nodes" use case.
+
+Implementing dedicated nodes using taints and tolerations is straightforward: in essence, a node that
+is dedicated to group A gets taint `dedicated=A` and the pods belonging to group A get
+toleration `dedicated=A`. (The exact syntax and semantics of taints and tolerations are
+described later in this doc.) This keeps all pods except those belonging to group A off of the nodes.
+This approach easily generalizes to pods that are allowed to
+schedule into multiple dedicated node groups, and nodes that are a member of multiple
```

Comment:
> Also note that it wouldn't be possible to put both dedicated=A and dedicated=B labels on the node to ensure affinity for dedicated pods of groups A and B.

Yes; to work around this I guess you could change the labels, e.g. some machines are `dedicated=A`, some machines are `dedicated=B`, and some machines are `dedicated=AandB`. This makes the tolerations more complicated (instead of tolerating `dedicated=A` you tolerate `dedicated=A` and `dedicated=AandB` but it still seems workable.

Anyway, I don't feel strongly one way or the other. I added a comment saying multiple taints with the same key are not allowed, but I'm also fine removing it.